prog/vm/nexus.rb

```diff
@@ -198,12 +198,19 @@ def allocation_dataset
         vm_host.used_cores + ? <= least(vm_host.total_cores, vm_host.total_mem_gib / ?)
         AND vm_host.used_hugepages_1g + ? <= vm_host.total_hugepages_1g
         AND vm_host.allocation_state = 'accepting'
-        AND vm_host.location = ?
         AND vm_host.arch = ?
-      ORDER BY random()
     SQL
 
-    DB[device_allocation_query, vm.cores, vm.mem_gib_ratio, vm.mem_gib, vm.location, vm.arch]
+    query_param_list = [vm.cores, vm.mem_gib_ratio, vm.mem_gib, vm.arch]
+
+    if vm.location != "github-runners"
+      device_allocation_query += " AND vm_host.location = ?"
+      query_param_list.append(vm.location)
+    end
+
+    device_allocation_query += " ORDER BY random()"
```

Comment:
I wonder if this can be slightly more elegant, I tried something like:

```ruby
DB[:pg_stat_activity].from_self.order(:xact_start).sql
```

But I don't think the juice is worth the squeeze, the query that comes out is ugly.  I wonder, though, if this can be more artful using interpolation `#{foo}` rather than catenation (`+=`)?

I'm generally becoming less happy with the amount of textual slashing here to generate SQL, but it's not  something for this patch to fix.

@enescakir you are pretty skilled with Sequel, any ideas?

Reply:
We might use something similar to the filters in authorization. It uses some kind of interpolation.
https://github.com/ubicloud/ubicloud/blob/3491ad72f85352ccf7942eab56c9da2d5ab48867/lib/authorization.rb#L39

Reply:
> We might use something similar to the filters in authorization. It uses some kind of interpolation.
> 
> https://github.com/ubicloud/ubicloud/blob/3491ad72f85352ccf7942eab56c9da2d5ab48867/lib/authorization.rb#L39

that does look rather appealing I think

Reply:
Today was holiday here & I'm out of office next week, so someone else please apply the feedback. You should be able to merge it after downloading github runner images to all hosts during week.